modules/ROOT/pages/kernel-args.adoc

```diff
@@ -1,10 +1,49 @@
 = Modifying Kernel Arguments
 
-Currently to change kernel arguments, you must script a systemd service which runs `rpm-ostree kargs --reboot`. The command supports appending kernel arguments (via `--append KEY[=VAL]`), deleting them (via `--delete KEY[=VAL]`), and replacing them (via `--replace KEY[=VALUE]`).
+Kernel arguments changes are managed by `rpm-ostree` via the https://www.mankier.com/1/rpm-ostree[`rpm-ostree kargs`] sub command. Changes are applied to a new deployment and a reboot is necessary for those to take effect.
 
-NOTE: In the future, we will have a more Ignition-friendly method of doing this with stronger guarantees. See https://github.com/coreos/ignition/issues/1051[this upstream issue] for more information.
+.Append kernel argument; useful with e.g. console= that can be used multiple times. empty value for an argument is allowed
+[source,bash]
+----
+$ sudo rpm-ostree kargs --append=KEY=VALUE
+----
+
+.Replace existing kernel argument, the user is also able to replace an argument with KEY=VALUE if only one value exist for that argument
+[source,bash]
+----
+$ sudo rpm-ostree kargs --replace=KEY=VALUE=NEWVALUE
+----
+
+.Delete a specific kernel argument key/val pair or an entire argument with a single key/value pair
+[source,bash]
+----
+$ sudo rpm-ostree kargs --delete=KEY=VALUE
+----
+
+.Use an editor to modify the kernel arguments
+[source,bash]
+----
+$ sudo rpm-ostree kargs --editor
+----
+
+.Exemple: Update an existing system from cgroupsv1 to cgroupsv2 and immediately reboot
 
-Here's an example which drops the default `systemd.unified_cgroup_hierarchy=0` kernel argument so that the machine uses cgroups v2:
+[source,bash]
+----
+$ sudo rpm-ostree --delete=systemd.unified_cgroup_hierarchy --reboot
+----
```

Comment:
This seems like a bad example now because we're defaulting to cgroups v2 so `systemd.unified_cgroup_hierarchy=0` won't be on the kernel command line. Is there another example we can use?

Reply:
We can add another example but we still need a place to put this one for all our existing users that will have to manually update their systems.

Reply:
Added more examples!

Reply:
> We can add another example but we still need a place to put this one for all our existing users that will have to manually update their systems.

Good point. I didn't think of that.